File: tests/fixtures/go/basic-crypto/pkg/hash/sha.go
package hash

import (
	"crypto/sha256"
	"crypto/sha512"
	"hash"
)

// HashSHA256 hashes data with SHA-256
func HashSHA256(data []byte) []byte {
	h := sha256.New()
	h.Write(data)
	return h.Sum(nil)
}

// HashSHA512 hashes data with SHA-512
func HashSHA512(data []byte) []byte {
	h := sha512.New()
	h.Write(data)
	return h.Sum(nil)
}

// GetHasher returns a hasher based on algorithm name
func GetHasher(algorithm string) hash.Hash {
	switch algorithm {
	case "sha256":
		return sha256.New()
	case "sha512":
		return sha512.New()
	default:
		return sha256.New()
	}
}

File: tests/fixtures/python/basic-crypto/hash/sha.py
"""SHA hashing functions."""

import hashlib


def hash_sha256(data: bytes) -> bytes:
    """Hash data with SHA-256."""
    h = hashlib.sha256()
    h.update(data)
    return h.digest()


def hash_sha512(data: bytes) -> bytes:
    """Hash data with SHA-512."""
    h = hashlib.sha512()
    h.update(data)
    return h.digest()


def get_hasher(algorithm: str):
    """Return a hasher based on algorithm name."""
    if algorithm == "sha256":
        return hashlib.sha256()
    elif algorithm == "sha512":
        return hashlib.sha512()
    else:
        return hashlib.sha256()
